File: Benchmarking/stats/euler15/euler15.java
import java.util.*;
import java.math.*;

public class euler15
{
	public static void main(String[] args)
	{
		int siz = Integer.parseInt(args[0]);
		System.out.println(calc(siz));
	}	
	
	private static BigInteger calc(int siz)
	{
		ArrayList<BigInteger> tmp = new ArrayList<>();
		for(int i = 0; i <= siz; i++)
		{
			tmp.add(BigInteger.valueOf(1));
		}
		ArrayList<ArrayList<BigInteger>> mat = new ArrayList<>();
		for(int i = 0; i <= siz; i++)
		{
			mat.add(new ArrayList<>(tmp));
		}
		
		for(int j = siz-1; j >= 0; j--)
		{
			for(int i = siz-1; i >= 0; i--)
			{
				mat.get(i).set(j, mat.get(i).get(j+1).add(mat.get(i+1).get(j)));
			}
		}
		
		return mat.get(0).get(0);
	}
}

File: Benchmarking/stats/euler15/euler15.jl
#=
Starting in the top left corner of a 2×2 grid, and only being able to move to
the right and down, there are exactly 6 routes to the bottom right corner.
How many such routes are there through a 20×20 grid?
=#
#Answer: 137846528820

#Terminal: julia euler15.jl 20
siz = parse(Int, ARGS[1])

function calc(siz)
  mat = ones(UInt128, siz + 1, siz + 1)

  for j = siz : -1 : 1
    for i = siz : -1 : 1
      mat[i, j] = mat[i, j + 1] + mat[i + 1, j]
    end
  end
  return mat[1][1]
end

println(calc(siz))
